src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -47,43 +51,57 @@ public override void Initialize(IHostContext hostContext)
             _agentServer = HostContext.GetService<IAgentServer>();
         }
 
-        public void UpdateAgentSetting(AgentSettings settings)
+        public void ReadSettings(AgentSettings settings)
         {
-            // No implementation required
         }
 
         public string GetServerUrl(CommandSettings command)
         {
             return command.GetUrl();
         }
 
-        public async Task<int> GetPoolId(CommandSettings command)
+        public async Task<AgentSettings> GetPoolId(CommandSettings command)
         {
             int poolId = 0;
             string poolName;
 
             poolName = command.GetPool();
             poolId = await GetPoolIdAsync(poolName);
             Trace.Info($"PoolId for agent pool '{poolName}' is '{poolId}'.");
- 
-            return poolId;
+
+            return new AgentSettings() { PoolId = poolId};
         }
 
         public string GetFailedToFindPoolErrorString() => StringUtil.Loc("FailedToFindPool");
 
-        public Task<TaskAgent> UpdateAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public string GetAgentWithSameNameAlreadyExistErrorString(AgentSettings agentSettings)
         {
-            return _agentServer.UpdateAgentAsync(poolId, agent);
+            return StringUtil.Loc("AgentWithSameNameAlreadyExistInPool", agentSettings.PoolId, agentSettings.AgentName);
         }
 
-        public Task<TaskAgent> AddAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public Task<TaskAgent> UpdateAgentAsync(AgentSettings agentSettings, TaskAgent agent, CommandSettings command)
+        {
+            return _agentServer.UpdateAgentAsync(agentSettings.PoolId, agent);
+        }
+
+        public Task<TaskAgent> AddAgentAsync(AgentSettings agentSettings, TaskAgent agent, CommandSettings command)
         { 
-            return _agentServer.AddAgentAsync(poolId, agent);
+            return _agentServer.AddAgentAsync(agentSettings.PoolId, agent);
         }
 
-        public Task DeleteAgentAsync(int agentPoolId, int agentId)
+        public async Task DeleteAgentAsync(AgentSettings agentSettings)
         {
-            return _agentServer.DeleteAgentAsync(agentPoolId, agentId);
+            string currentAction = StringUtil.Loc("UninstallingService");
```

Comment:
it's weird to have this string here. why call DeleteAgent will print out UninstallService?